jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/impl/HiveSqlGenerationVisitor.java

```diff
@@ -0,0 +1,255 @@
+package com.exasol.adapter.dialects.impl;
+
+import com.exasol.adapter.dialects.SqlDialect;
+import com.exasol.adapter.dialects.SqlGenerationContext;
+import com.exasol.adapter.dialects.SqlGenerationVisitor;
+import com.exasol.adapter.jdbc.ColumnAdapterNotes;
+import com.exasol.adapter.metadata.ColumnMetadata;
+import com.exasol.adapter.metadata.DataType;
+import com.exasol.adapter.sql.*;
+import com.google.common.base.Joiner;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by np on 12/30/2016.
+ */
+public class HiveSqlGenerationVisitor extends SqlGenerationVisitor {
+
+
+    public HiveSqlGenerationVisitor(SqlDialect dialect, SqlGenerationContext context) {
+        super(dialect, context);
+    }
+
+
+    @Override
+    public String visit(SqlSelectList selectList) {
+        List<String> selectListElements = new ArrayList<>();
+        SqlStatementSelect select = (SqlStatementSelect) selectList.getParent();
+        if (selectList.isSelectStar()) {
+            if (selectListRequiresCasts(selectList)) {
+                // Do as if the user has all columns in select list
+                int columnId = 0;
+                for (ColumnMetadata columnMeta : select.getFromClause().getMetadata().getColumns()) {
+                    SqlColumn sqlColumn = new SqlColumn(columnId, columnMeta);
+                    if (columnMeta.getType().toString().equals("VARCHAR(2000000) UTF8")) {
+                        selectListElements.add("base64(" + super.visit(sqlColumn) + ")");
+                    } else {
+                        selectListElements.add(super.visit(sqlColumn));
+                    }
+                    ++columnId;
+
+                }    }
+            else {
+                    selectListElements.add("*");
+                }
+
+        } else {
+           // this if is added because of the RAND function(otherwise if you select only rand... you will get an exception
```

Comment:
Not sure what is meant by the comment, because I don't see a check for rand - only the following if, or the whole else section?